.github/workflows/run-api-tests.yml

```diff
@@ -30,6 +32,14 @@ jobs:
 
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v1
+
+      - name: Publish docker image
```

Comment:
I think we should publish it after the tests passed. Unless you want to publish those images for debugging.

I would also use https://github.com/docker/login-action#docker-hub to login as it will make sure we do not leak any credentials.

Reply:
Yes, for feature branches it's usually useful to publish even if tests fail. Eg when we were working on deduplication, NTI on feature branches, I was writing tests as devs were fixing bugs, so there was a good chance of failing tests :) 

Creds are not leaked with --password-stdin, but I can use their workflow if you prefer. 
![image](https://user-images.githubusercontent.com/12664326/145369517-8ba71092-7644-46ac-ab42-f27ef9e9232d.png)

Reply:
I prefer the login-action 😄

Reply:
Changed a one-liner to 6-liner 😄

Reply:
it was at least a 3-liner ;)